firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -203,7 +203,9 @@ private void recalculateAndSaveOverlays(Map<DocumentKey, MutableDocument> docs)
     // along the way.
     for (MutationBatch batch : batches) {
       for (DocumentKey key : batch.getKeys()) {
-        FieldMask mask = batch.applyToLocalView(docs.get(key), masks.get(key));
+        FieldMask mask =
+            masks.containsKey(key) ? masks.get(key) : FieldMask.fromSet(new HashSet<>());
```

Comment:
Can we create an EMPTY_MASK sentinel? The missing mask is the default case and we create two objects each time.

Reply:
Done.